dashboard/test/ui/features/ed_programs/pd/application_detail_view.feature

```diff
@@ -0,0 +1,25 @@
+@dashboard_db_access
```

Comment:
this can be slow and therefore flaky. can you open a follow-up item to eliminate this, and we can discuss at midsprint? the work would be to define an api in test_controller.rb, and then hit that api to do the necessary setup rather than access the dashboard db directly. I am guessing it is just this that needs it: https://github.com/code-dot-org/code-dot-org/blob/46b26f2d5eb43dfb02fc3cb4416a14c34e62d76d/dashboard/test/ui/features/step_definitions/pd.rb#L1-L9

Reply:
Ah, I see. Yeah, I used that because that's what I saw in `teacher_application_dashboard.feature` and I figured it needed it to create fake applications.

I'm most worried about this interacting with `teacher_application_dashboard.feature` –– I only need one application, but don't want to create a new one that could mess up `teacher_application_dashboard.feature`. The `creating some fake applications of each type and status` step doesn't create new apps if they already exist:

https://github.com/code-dot-org/code-dot-org/blob/9b49f259b8f1b5646dd4a583b7efd457dc45ccaf/dashboard/test/ui/features/step_definitions/pd.rb#L162-L165

Anyways, we can chat at midsprint. Shall I wait to merge this, or create a separate work item to address?

Reply:
good point about tests running simultaneously. to that point, simply cleaning up at the beginning of each scenario wouldn't work well.

I won't have time to dig in til after demos, but here are a couple half-baked ideas:
1. the `@only_one_browser` flag, though this would still mean you have two test runs (one eyes, one non)
2. can you identify the application, and find it in the browser window, by something other than assuming it is the first and only application?

Reply:
Just checked the `teacher_application_dashboard` test –– we're ignoring the regions that show how many apps are created, and it doesn't matter which one gets clicked, so long as it has the right status. So, I think we're safe here